src/test_workflow/perf_test/perf_test_cluster.py

```diff
@@ -13,32 +13,20 @@
 
 class PerfTestCluster(TestCluster):
     """
-    Represents a performance test cluster. This class deploys the opensearch bundle with CDK and returns the private IP.
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK. Supports both single
+    and multi-node clusters
     """
 
-    def __init__(self, bundle_manifest, config, stack_name, security, current_workspace):
+    def __init__(self, bundle_manifest, config, stack_name, cluster_config, current_workspace):
         self.manifest = bundle_manifest
-        self.work_dir = os.path.join(current_workspace, "opensearch-cluster", "cdk", "single-node")
+        self.work_dir = os.path.join(current_workspace, "opensearch-cluster", "cdk",
```

Comment:
You should subclass `PerfTestCluster` into single vs. multi-node. All the methods such as `create_endpoint` and `setup_cdk_params` have different implementations in those cases.

Reply:
done.